libs/@guardian/libs/src/performance/startPerformanceMeasure.ts

```diff
@@ -0,0 +1,46 @@
+import type { TeamName } from '../logger/@types/logger';
+import type {} from './@types/measure';
+import { serialise } from './serialise';
+
+/** For browser which do not fully support the performance API */
+const fallback: ReturnType<typeof startPerformanceMeasure> = {
+	endPerformanceMeasure: () => -1,
+};
+
+/**
+ * Helper to measure the duration between two events.
+ *
+ * Once ended, measures are appended to the list of `PerformanceMeasure`
+ * and can be retrieved with `performance.getEntriesByType('measure')`
+ *
+ * @example
+ * const { endPerformanceMeasure } = startPerformanceMeasure('dotcom', 'fetch');
+ * await fetch('https://www.theguardian.com/uk.json');
+ * const duration = endPerformanceMeasure();
+ */
+export const startPerformanceMeasure = (
+	team: TeamName,
+	name: string,
+	action?: string,
+): { endPerformanceMeasure: () => number } => {
+	if (!('getEntriesByName' in window.performance)) {
+		return fallback;
+	}
+
+	const options = {
+		start: performance.now(),
+	} satisfies PerformanceMeasureOptions;
+
+	const endPerformanceMeasure = () => {
+		const measureName = serialise(team, name, action);
```

Comment:
should we do this in start, before calling `performance.now()`? otherwise the time to serialise will be included in the duration